source/configuration/modules/imfile.rst

```diff
@@ -1,570 +1,408 @@
-object and keep LF escaping turned on. 
+By default, it's set to *on* if *file* contains *wildcards*, otherwise default is *off*.
 
-Legacy Configuration Directives
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+==================  ====================
+trimlineoverbytes:  "number"
+Default:            "0 (never truncate)"
+Available since:    8.17.0
+==================  ====================
 
-.. index:: 
-   single: imfile; $InputFileName
-.. function:: $InputFileName /path/to/file
+**This option can be used only when *readMode* is 0 or 2**.
+This is used to tell rsyslog to truncate the line which length is greater
+than specified bytes. If it is positive number, rsyslog truncate the line
+at specified bytes. 
 
-   equivalent to "file"
+================  ========
+freshstarttail:   "on|off"
+Default:          "off"
+Available since:  8.18.0
+================  ========
 
-.. index:: 
-   single: imfile; $InputFileTag
-.. function:: $InputFileTag tag:
+If set to *off*, rsyslog will read files from *last read position* as defined in State files.
+If set to *on*, rsyslog will seek the file reading position at the end/tail, processing new events written since that moment.
+It's very useful when deploying rsyslog to a large number of servers, because it allows to discard old logs.
 
-   equivalent to: "tag"
-   you would like to see the colon after the tag, you need to specify it
-   here (as shown above).
+=====================  ==============================
+persiststateinterval:  "number"
+Default:               "0 (end rsyslog execution)"
+Legacy:                $InputFilePersistStateInterval
+=====================  ==============================
 
-.. index:: 
-   single: imfile; $InputFileStateFile
-.. function:: $InputFileStateFile name-of-state-file
+Specifies how often (lines read) the state file shall be written when processing the input file. 
 
-   equivalent to: "StateFile"
+If set to 0, it means a new state file is only written when the monitored files are closed (end of rsyslogd execution).
+Any other value means that the state file is written when *number* lines have been processed.
 
-.. index:: 
-   single: imfile; $InputFileFacility
-.. function:: $InputFileFacility facility
+This setting can be used to guard against message duplication due to fatal errors (like power fail).
+Note that this setting affects imfile performance, especially when set to a low value. 
 
-   equivalent to: "Facility"
+Multiline parameters
+====================
 
-.. index:: 
-   single: imfile; $InputFileSeverity
-.. function:: $InputFileSeverity severity
+=========  =======
+readmode:  "0|1|2"
+Default:   "0"
+=========  =======
 
-   equivalent to: "Severity"
+Provides support for *simple* multi-line processing. Incompatible with *startmsg.regex*. 
 
-.. index:: 
-   single: imfile; $InputRunFileMonitor
-.. function:: $InputRunFileMonitor
+* 0 - line: each line is a new message.
```

Comment:
with the addition of the startmsg.regex, it's probably a good idea to mark this as depriciated and list what the appropriate regex would be to use instead.

In the future, the internal implementation of these may change to do this anyway.